buffer/src/main/java/io/netty/buffer/AbstractUnpooledSlicedByteBuf.java

```diff
@@ -214,11 +214,13 @@ protected long _getLongLE(int index) {
         return unwrap().getLongLE(idx(index));
     }
 
+    @SuppressWarnings("deprecation")
     @Override
     public ByteBuf duplicate() {
-        final ByteBuf duplicate = unwrap().slice(adjustment, length());
-        duplicate.setIndex(readerIndex(), writerIndex());
-        return duplicate;
+        final ByteBuf unwrapped = unwrap();
+        return (unwrapped instanceof AbstractByteBuf ?
+                new UnpooledDuplicatedByteBuf((AbstractByteBuf) unwrapped) :
+                new DuplicatedByteBuf(unwrapped)).setIndex(idx(readerIndex()), idx(writerIndex()));
```

Comment:
I was using why we return a different `ByteBuf` implementation depending on the type and can't just call `unwrap.duplicate().setIndex(...)` here.